### DISCUSSION TITLE
"Paste and Go" right click option for the URL bar

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#823] [Ideas]
### What feature would you like to see?

In Firefox when you have something in the clipboard (URL or text), you have an option "paste and go" when right-clicking on the URL bar. Clicking it will either send you to the URL from the clipboard, or it will search if the text isn't a URL

### What platform are you seeing the problem on?

Linux

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/485_

**Comment by SomeDudeBrosky** [comment]
This seems like a good idea, however the naming seems unclear and should be discussed further

**Reply by ch4og** [reply]
This is default naming used in firefox so ig it's ok
![image](https://github.com/user-attachments/assets/ffb5dfe1-66a2-4df3-a298-5375002ae1b0)

**Comment by tunalad** [comment]
It would also be nice if there was that option when right-clicking on the tab sidebar too actually

**Comment by Dean-Corso** [comment]
Hi guys,

I did notice that the URL bar context menu with "paste and go" menu entry is missing in default run. When you now open the customize Toolbar tab and close it with ESC then the menu entry "paste and go" is there and choose able. Why? I don't want to open and close the CT tab just to enable that menu entry so it must be some bug.

Otherwise a simple button for that function "paste and go" next to URL bar would be nice to prevent to open context menu / choose menu entry manually each time. I'm using "paste and go" pretty often and would simplify that process by just clicking on a button.

I fond a add on called "[Multiple Paste and Go Button](https://addons.mozilla.org/en-US/firefox/addon/multiple-paste-and-go-button/)" what can do that (also multiple URLs from clipboard) but it does just create NEW tabs and does not use the selected tab I want to paste & go.

A better working add on I found what can do that is "[Paste and Go Key](https://addons.mozilla.org/en-US/firefox/addon/paste-and-go-key/)". So would be a good idea to add such an button directly into ZEN browser itself without using any external add on for that.

**Reply by tunalad** [reply]
> When you now open the customize Toolbar tab and close it with ESC then the menu entry "paste and go" is there and choose able.

I didn't expect this to work, but it did. I don't know either why they had to hide it since I also heavily rely on that button, and I assume many do too
